### PR TITLE
SQL-1036: Add helper function to verify SQL diagnostic records

### PR DESCRIPTION
**Pull request by bucaojit**
Refactored `odbc/src/api/driver_connect_tests.rs` and moved it to the integration test module.  Modified it to use `Handle` instead of `MongoHandle`.
Added helper function `verify_sql_diagnostics` to obtain the SQLGetDiagRec output and compare with sql state, message text, and native error.

### CODE REVIEW

File: odbc/Cargo.toml [hunk 1]
```diff
@@ -32,4 +32,4 @@ thiserror = "1"
 
 [lib]
 name = "mongoodbc"
-crate-type = ["cdylib"]
+crate-type = ["cdylib","lib"]
```
Comment (bucaojit) [inline odbc/Cargo.toml:35]: I added `lib` here so that `mongoodbc` could be imported from the odbc/tests/ directory.

**Review by mattChiaravalloti — APPROVED** [review]

**Comment by pmeredit** [comment]
@bucaojit I suspect that the linker errors are due to building it as both a lib and a dynlib. It's probably linking against the static names and still trying to dynamically link in the dynlib.

**Comment by bucaojit** [comment]
Thanks for the help @pmeredit on the linker issue.  
I moved the tests that used the `odbc-api` library (test_invalid_connection and test_default_connection) to the integration_test directory and they link and run as seen [here](https://spruce.mongodb.com/task/mongosql_odbc_driver_windows_64_integration_test_patch_17c7edb856a7e025fe441cb9565772d1cbbfd0c2_6348d8681e2d1753c3f511e2_22_10_14_03_32_57/logs?execution=0).

**Review by pmeredit — APPROVED** [review]
LGTM

**Comment by bucaojit** [comment]
evergreen merge

**Comment by bucaojit** [comment]
evergreen merge

**Pull request merged**